client/lib/catch-js-errors/index.js

```diff
@@ -11,24 +11,27 @@ export default class ErrorLogger {
 
 		if ( ! window.onerror ) {
 			TraceKit.report.subscribe( errorReport => {
+				const error = {
+					message: errorReport.message,
+					url: document.location.href
+				};
+
 				if ( Array.isArray( errorReport.stack ) ) {
-					errorReport.stack.forEach( report => Object.keys( report ).forEach( key => {
+					const trace = errorReport.stack.slice( 0, 10 );
+					trace.forEach( report => Object.keys( report ).forEach( key => {
 						if ( key === 'context' && report[ key ] ) {
-							report[ key ] = JSON.stringify( report[ key ] ).substring( 0, 512 );
+							report[ key ] = JSON.stringify( report[ key ] ).substring( 0, 256 );
```

Comment:
I admit to knowing nothing about this module however I do see that `report[ key ]` is now being truncated to `256` here but is still allowed to be up to `512` bytes in the line below. Just noting it incase that was an oversight.